casper/src/test/scala/coop/rchain/casper/ValidateTest.scala

```diff
@@ -628,41 +628,18 @@ class ValidateTest
 
   "Justification regression validation" should "return valid for proper justifications and justification regression detected otherwise" in withStorage {
     implicit blockStore => implicit blockDagStorage =>
-      val validators = Vector(
-        generateValidator("Validator 1"),
-        generateValidator("Valdiator 2")
-      )
-      val bonds = validators.zipWithIndex.map {
+      val v0 = generateValidator("Validator 1")
+      val v1 = generateValidator("Validator 2")
+      val bonds = List(v0, v1).zipWithIndex.map {
         case (v, i) => Bond(v, 2L * i.toLong + 1L)
       }
 
-      def latestMessages(messages: Seq[BlockMessage]): Map[Validator, BlockHash] =
-        messages.map(b => b.sender -> b.blockHash).toMap
-
-      def createValidatorBlock[F[_]: Monad: Time: BlockStore: IndexedBlockDagStorage](
-          parents: Seq[BlockMessage],
-          genesis: BlockMessage,
-          justifications: Seq[BlockMessage],
-          validator: Int
-      ): F[BlockMessage] =
-        for {
-          deploy <- ConstructDeploy.basicProcessedDeploy[F](0)
-          result <- createBlock[F](
-                     parents.map(_.blockHash),
-                     genesis,
-                     creator = validators(validator),
-                     bonds = bonds,
-                     deploys = Seq(deploy),
-                     justifications = latestMessages(justifications)
-                   )
-        } yield result
-
       for {
         b0 <- createGenesis[Task](bonds = bonds)
-        b1 <- createValidatorBlock[Task](Seq(b0), b0, Seq(b0, b0), 0)
-        b2 <- createValidatorBlock[Task](Seq(b1), b0, Seq(b1, b0), 0)
-        b3 <- createValidatorBlock[Task](Seq(b0), b0, Seq(b2, b0), 1)
-        b4 <- createValidatorBlock[Task](Seq(b3), b0, Seq(b2, b3), 1)
+        b1 <- createValidatorBlock[Task](Seq(b0), b0, Seq(b0, b0), v0, bonds)
```

Comment:
you should split the refactoring that extracts `createValidatorBlock` into a separate commit